cdap-ui/app/cdap/components/DataPrep/TopPanel/AddToPipelineModal.js

```diff
@@ -431,6 +430,56 @@ export default class AddToHydratorModal extends Component {
       });
   }
 
+  applyDirective(directive) {
+    execute([directive])
+      .subscribe(
+        () => {
+          this.setState({
+            error: null,
+            loading: true,
+            schema: []
+          });
+          setTimeout(() => {
```

Comment:
nit: Is it safer to have the `generateLinks` as second argument to `setState` since its async